src/offset/fixed.rs

```diff
@@ -16,16 +17,50 @@ use crate::oldtime::Duration as OldDuration;
 use crate::DateTime;
 use crate::Timelike;
 
+const OFFSET_NORMAL: i32 = 1;
+const OFFSET_UNKNOWN: i32 = 2;
+
 /// The time zone with fixed offset, from UTC-23:59:59 to UTC+23:59:59.
 ///
 /// Using the [`TimeZone`](./trait.TimeZone.html) methods
 /// on a `FixedOffset` struct is the preferred way to construct
 /// `DateTime<FixedOffset>` instances. See the [`east_opt`](#method.east_opt) and
 /// [`west_opt`](#method.west_opt) methods for examples.
-#[derive(PartialEq, Eq, Hash, Copy, Clone)]
+#[derive(Eq, Copy, Clone)]
 #[cfg_attr(feature = "rkyv", derive(Archive, Deserialize, Serialize))]
 pub struct FixedOffset {
-    local_minus_utc: i32,
+    // Encodes an offset in seconds, with the value shifted two bits to the left.
+    // The remaining bits encode flags:
+    // - `OFFSET_NORMAL` sets one flag to a non-zero value, so we can encode it in a `NonZeroI32`
+    //   and get niche optimization.
+    // - `OFFSET_UNKNOWN` to preserve the difference RFC 2822 and RFC 3339 make between an offset
+    //    of +0:00 and -0:00.
+    // Advantage of this encoding is that it only takes a single shift right to get offset in
+    // seconds.
+    local_minus_utc: NonZeroI32,
```

Comment:
Can you make this code comment a docstring?
In the docstring, would you tell the user what helper functions they should call to extract various infos?  Can you provide those helper functions? Even if it's only chrono developers that need to know, it's still worthwhile to write down.
Otherwise I might assume I have to do the bit shifts and masks myself... and those are programming approaches I would like everyone to move on from. 😬

Reply:
> honestly, I'd prefer to have discrete struct members for each piece of information embedded here. I'm not persuaded the runtime efficiency-savings is worth the mental overhead.

This is somewhat comparable to `NaiveDate`, which fits a year, ordinal, leap-year flag and weekday flags in an i32.

I agree that using individual struct members is usually better. But chrono's `DateTime<FixedOffset>` is a pretty fundamental type in the rust ecosystem. It should not increase in size without very good reason.

> Also, I'm not sure, but I'm concerned it might be a problem on less-common Big Endian platforms (I don't recall where and when Rust gives guarantees around bit endianness).

In safe Rust, we do not have to worry about this. As soons as we start transmuting it becomes messy.